# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/ffmpeg/FFmpegDecoderModule.h
# Commit: f9337268523c
# Full Hash: f9337268523c6607228fda7fa659fd5a3d939dad
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - P1E. Don't use libvpx in the RDD process if video contains an alpha plane. r=mattwoodrow
#   
#   Until bug 1668840 is fixed, we can't allocate a SharedRGBImage without a PImageBridge
#   
#   Depends on D92233
# ==============================================================================

diff -r 7a990fa02246 -r f9337268523c dom/media/platforms/ffmpeg/FFmpegDecoderModule.h
--- a/dom/media/platforms/ffmpeg/FFmpegDecoderModule.h	Tue Oct 20 23:30:04 2020 +0000
+++ b/dom/media/platforms/ffmpeg/FFmpegDecoderModule.h	Tue Oct 20 23:30:18 2020 +0000
@@ -31,13 +31,6 @@
 
   already_AddRefed<MediaDataDecoder> CreateVideoDecoder(
       const CreateDecoderParams& aParams) override {
-    // Temporary - forces use of VPXDecoder when alpha is present.
-    // Bug 1263836 will handle alpha scenario once implemented. It will shift
-    // the check for alpha to PDMFactory but not itself remove the need for a
-    // check.
-    if (aParams.VideoConfig().HasAlpha()) {
-      return nullptr;
-    }
     RefPtr<MediaDataDecoder> decoder = new FFmpegVideoDecoder<V>(
         mLib, aParams.VideoConfig(), aParams.mKnowsCompositor,
         aParams.mImageContainer,
@@ -56,8 +49,28 @@
 
   bool SupportsMimeType(const nsACString& aMimeType,
                         DecoderDoctorDiagnostics* aDiagnostics) const override {
-    AVCodecID videoCodec = FFmpegVideoDecoder<V>::GetCodecId(aMimeType);
-    AVCodecID audioCodec = FFmpegAudioDecoder<V>::GetCodecId(aMimeType);
+    UniquePtr<TrackInfo> trackInfo = CreateTrackInfoWithMIMEType(aMimeType);
+    if (!trackInfo) {
+      return false;
+    }
+    return Supports(SupportDecoderParams(*trackInfo), aDiagnostics);
+  }
+
+  bool Supports(const SupportDecoderParams& aParams,
+                DecoderDoctorDiagnostics* aDiagnostics) const override {
+    const auto& trackInfo = aParams.mConfig;
+    const nsACString& mimeType = trackInfo.mMimeType;
+
+    // Temporary - forces use of VPXDecoder when alpha is present.
+    // Bug 1263836 will handle alpha scenario once implemented. It will shift
+    // the check for alpha to PDMFactory but not itself remove the need for a
+    // check.
+    if (VPXDecoder::IsVPX(mimeType) && trackInfo.GetAsVideoInfo()->HasAlpha()) {
+      return false;
+    }
+
+    AVCodecID videoCodec = FFmpegVideoDecoder<V>::GetCodecId(mimeType);
+    AVCodecID audioCodec = FFmpegAudioDecoder<V>::GetCodecId(mimeType);
     if (audioCodec == AV_CODEC_ID_NONE && videoCodec == AV_CODEC_ID_NONE) {
       return false;
     }